sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -527,6 +528,17 @@ protected Symbol visitInPredicate(InPredicate node, ExpressionAnalysisContext co
             return process(arrayComparisonExpression, context);
         }
 
+        @Override
+        protected Symbol visitArraySubQueryExpression(ArraySubQueryExpression node, ExpressionAnalysisContext context) {
+            context.forceReturnMultipleRows(true);
+            Symbol argument = process(node.subqueryExpression(), context);
+            context.forceReturnMultipleRows(false);
```

Comment:
I think it's save to leave it true for the entire operation no?

Reply:
what if there is another array(subquery) operation in the where clause or in another nested query? Would it still be safe?

Reply:
This is the context for this expression. The nested expressions would have their own analysis. @mfussenegger do you have anything to add?